saleor/checkout/webhooks/shipping_list_methods_for_checkout.py

```diff
@@ -0,0 +1,90 @@
+from collections.abc import Iterable
+from typing import Optional
+
+from django.conf import settings
+
+from ...core.middleware import Requestor
+from ...permission.enums import ShippingPermissions
+from ...shipping.interface import ShippingMethodData
+from ...webhook.base import SyncWebhookBase
+from ...webhook.const import CACHE_TIME_SHIPPING_LIST_METHODS_FOR_CHECKOUT
+from ...webhook.utils import get_webhooks_for_event
+from ..models import Checkout
+
+
+class ShippingListMethodsForCheckout(SyncWebhookBase):
+    description = "Fetch external shipping methods for checkout."
+    event_type = "shipping_list_methods_for_checkout"
+    name = "List shipping methods for checkout"
+    permission = ShippingPermissions.MANAGE_SHIPPING
+    subscription_type = (
+        "saleor.graphql.checkout.subscriptions.ShippingListMethodsForCheckout"
+    )
+
+    @classmethod
+    def list_shipping_methods(
+        cls,
+        checkout: Checkout,
+        requestor: Optional[Requestor] = None,
+        allow_replica: bool = True,
+    ) -> Iterable[ShippingMethodData]:
+        """Fetch list of external shipping methods for checkout."""
+
+        from ...webhook.payloads import generate_checkout_payload
+        from ...webhook.transport.shipping import (
+            get_cache_data_for_shipping_list_methods_for_checkout,
```

Comment:
This is something related only to this webhook. I don't think we need to still keep this in `transport.shipping`. IMO this should be a part of this webhook implementation. After migrating all webhooks we should have the possibility to fully drop any custom logic from `transport` module and move them to webhook-related files.
(we can also handle this in a separate task, but let's make sure that we have todo point/task about this.)

Reply:
Yep, that's the plan to do it like that.